src/api_authentication_app_skeleton/spec/support/app_visitor.cr

```diff
@@ -0,0 +1,131 @@
+require "http/client"
+
+class AppVisitor
+  include Auth::ActionHelpers
+
+  CSRF_TOKEN = "statictesttoken"
+
+  getter! response
+  @response : HTTP::Client::Response? = nil
+
+  def visit(path : String, as_user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if as_user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", generate_token(as_user)
+    end
+
+    request = HTTP::Request.new("GET", path, headers)
+    @response = process_request(request)
+  end
+
+  def put(path : String, body : Hash(String, String), as_user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if as_user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", generate_token(as_user)
+    end
+
+    request_with_body("PUT", path, with_csrf_token(body), headers)
+  end
+
+  def post(path : String, body : Hash(String, String), as_user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if as_user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", generate_token(as_user)
+    end
+
+    request_with_body("POST", path, with_csrf_token(body), headers)
+  end
+
+  def delete(path : String, as_user : Authentic::PasswordAuthenticatable? = nil, headers : HTTP::Headers? = nil)
+    if as_user
+      headers ||= HTTP::Headers.new
+      headers.add "Authorization", generate_token(as_user)
+    end
+
+    request = HTTP::Request.new("DELETE", path, headers)
+    @response = process_request(request)
+  end
+
+  def response_body
+    body = @response.not_nil!.body
+    if body[0, 5] == "ERROR"
+      raise Exception.new(body)
+    elsif body == ""
+      JSON.parse("{}")
+    else
+      JSON.parse(body)
+    end
+  end
```

Comment:
I think `response_body` should return the raw_response and there should be another method called `response_json` that parses it. That way this method can work for HTML/XML or whatever people want